Mobile Buy SDK/Mobile Buy SDK/Payment Providers/BUYApplePayPaymentProvider.m

```diff
@@ -176,16 +190,32 @@ - (void)proceedWithApplePay
 
 	PKPaymentRequest *request = [self paymentRequest];
 	request.paymentSummaryItems = [self.checkout buy_summaryItemsWithShopName:self.shop.name];
-	PKPaymentAuthorizationViewController *controller = [[PKPaymentAuthorizationViewController alloc] initWithPaymentRequest:request];
-	if (controller) {
-		controller.delegate = self;
-		[self.delegate paymentProvider:self wantsControllerPresented:controller];
-	}
-	else {
-		if ([self.delegate respondsToSelector:@selector(paymentProvider:didFailWithError:)]) {
-			[self.delegate paymentProvider:self didFailWithError:nil];
+	if (SYSTEM_VERSION_LESS_THAN(@"10.0")) {
+		PKPaymentAuthorizationViewController *controller = [[PKPaymentAuthorizationViewController alloc] initWithPaymentRequest:request];
```

Comment:
Should try to avoid the code duplication here.  You should be able to have method that returns a Class (either PKPaymentAuthorizationViewController or PKPaymentAuthorizationController) that can be used here without having an if-else